core/src/main/java/ai/timefold/solver/core/api/domain/variable/CascadeUpdateElementShadowVariable.java

```diff
@@ -0,0 +1,32 @@
+package ai.timefold.solver.core.api.domain.variable;
+
+import static java.lang.annotation.ElementType.FIELD;
+import static java.lang.annotation.RetentionPolicy.RUNTIME;
+
+import java.lang.annotation.Retention;
+import java.lang.annotation.Target;
+
+/**
+ * A listener automatically sourced on a {@link PreviousElementShadowVariable} and {@link InverseRelationShadowVariable}.
+ * <p>
+ * Change shadow variables when the previous or inverse relation elements changes.
+ * <p>
+ * Automatically cascades change events to {@link NextElementShadowVariable} of a {@link PlanningListVariable}.
+ * <p>
+ * Important: it must only change the shadow variable(s) for which it's configured!
```

Comment:
```suggestion
 * Important: it must only change the shadow variable(s) for which it's configured.
```

We generally don't shout at people. :-)